eventcore-testing/src/lib.rs

```diff
@@ -0,0 +1,5 @@
+#![deny(warnings)]
```

Comment:
Add the full set of linting annotations that we have in the main library's lib.rs. The code quality constraints are the same.